cep-20.2.md

```diff
@@ -0,0 +1,650 @@
+    # rpaths_patcher: None
+```
+
+## Source section
+
+```yaml
+source: [SourceElement]
+```
+
+where the different source elements are defined as follows.
+
+### URL source
+
+```yaml
+# url pointing to the source tar.gz|zip|tar.bz2|...
+url: url
+# destination folder in work directory
+folder: path
+# hash of the file
+sha256: hex string
+# absolute or relative path from recipe file
+patches: [path]
+# removed
+# fn: string
+# md5: hex string
+# sha1: hex string
+```
+
+### Local source
+
+```yaml
+# file, absolute or relative to recipe file
+path: path
+# destination folder
+folder: path
+# absolute or relative path from recipe file
+patches: [path]
+```
+
+### Git source
+
+```yaml
+# URL to the git repository or path to local git repository
+git_url: url | path
+# revision to checkout to (commit or tag)
+git_rev: string
+git_depth: signed integer (defaults to -1 -> not shallow)
+```
+
+### Removed source definitions
+
+SVN and HG (mercury) source definitions are removed as they are not relevant anymore.
+
+## Requirements section
+
+```yaml
+requirements:
+  # build time dependencies, in the build_platform architecture
+  build: [MatchSpec]
+  # dependencies to link against, in the target_platform architecture
+  host: [MatchSpec]
+  # the section below is copied into the index.json and required at package installation
+  run: [MatchSpec]
+  # constrain optional packages
+  run_constrained: [MatchSpec]
+```
+
+## Test section
+
+Note: the test section also has a weird implicit behavior with `run_test.sh`, `run_test.bat` as well as `run_test.py` and `run_test.pl` script files
+that are run as part of the test.
+
+This is the current state of the Test section.
+
+```yaml
+test:
+  # files (from recipe directory) to include with the tests
+  files: [glob]
+  # files (from the work directory) to include with the tests
+  source_files: [glob]
+  # requirements at test time, in the target_platform architecture
+  requires: [MatchSpec]
+  # commands to execute
+  commands: [string]
+  # imports to execute with python (e.g. `import <string>`)
+  imports: [string]
+  # downstream packages that should be tested against this package
+  downstreams: [MatchSpec]
+```
+
+### Proposal for new test section:
+
+```
+test: [TestElement]
+```
+
+where the different test elements are defined as follows:
+
+#### Command test element
+
+```yaml
+# script to execute
+script: string | [string]
+# optional extra requirements
+extra_requirements:
+  # extra requirements with build_platform architecture (emulators, ...)
+  build: [MatchSpec]
+  # extra run dependencies
+  run: [MatchSpec]
+# extra files to add to the package for the test
+# Alternatively, use ${{ RECIPE_DIR }} or ${{ SOURCE_DIR }} and a single list?
+# or $RECIPE_DIR / $SRC_DIR ...
+files:
+  # files from $SRC_DIR
+  source: [glob]
+  # files from $RECIPE_DIR
+  recipe: [glob]
+```
+
+#### Import test element
+
+```yaml
+# list of imports to try
+imports: [string]
```

Comment:
As we move to an language agnostic world, should this be renamed to `python_imports` for clarity?

Reply:
Or we do

```yaml
python:
  imports:
    - mypkg
    - mypkg.foo
```

With the potential to add more keys, e.g. `pytest` or others?

Reply:
That could work, yes. I think `pytest` would be better handled as a command anyway, but maybe there are other Python specific checks to be made.

Reply:
I'm thinking renaming `imports` to `python` is sufficient (i.e., language keys are imports):

```
test:
  python:
    - mypkg
    - mypkg.foo
    - mypkg.foo:function
  rust:
    - mypkg
    - mypkg::foo
    - mypkg::foo::function
```

Anything more complex is probably better off in a script?